plugins/onepad/joystick.cpp

```diff
@@ -218,7 +218,7 @@ bool JoystickInfo::Init(int id)
 	// Keep the 4 hat buttons too (usb driver). (left pressure does not work with recent kernel). Moreover the pressure
 	// work sometime on half axis neg others time in fulll axis. So better keep them as button for the moment
 	u32 found_hack = devname.find(string("PLAYSTATION(R)3"));
-	if (found_hack != string::npos) {
+	if (found_hack != string::npos && (conf->options & PADOPTION_SIXAXIS_USB) == 0) {
```

Comment:
Is the `== 0` correct? If it's toggling the hack on and off ` (conf->options & PADOPTION_SIXAXIS_USB) == 0` means the hack is off so 19 buttons should be detected in that case.